### PR TITLE
WT-9792 Fix RTS to remove globally visible update of the data store update from the history store

### PR DESCRIPTION
**Pull request by kommiharibabu**
For a globally visible update, the start timestamp is reset to 0. The globally
visible updates present in the history store are missed to remove from the
history store when the same update is restored as a stable update in the
datastore due to a wrong start timestamp comparison. Adjust the check
to compare the stop timestamp instead of the start timestamp to remove
the globally visible updates from the history store.

Along with the above fix, let the history store will be scanned at the
end of RTS for any unstable updates to remove in all scenarios as it used
to do only during recovery.

### CODE REVIEW

File: src/txn/txn_rollback_to_stable.c [hunk 1]
```diff
@@ -1865,7 +1875,15 @@ __rollback_to_stable_btree_apply_all(WT_SESSION_IMPL *session, wt_timestamp_t ro
...
-    if (F_ISSET(S2C(session), WT_CONN_RECOVERING))
+    /*
+     * Performing eviction in parallel to a checkpoint can lead to situation where the history store
+     * have more updates than its corresponding data store. Performing history store cleanup at the
+     * end can able to remove any such unstable updates that are written to the history store.
+     *
+     * Do not perform the final pass on the history store in an in-memory configuration as it
+     * doesn't exist.
+     */
+    if (!F_ISSET(S2C(session), WT_CONN_IN_MEMORY))
```
Comment (quchenhao) [inline src/txn/txn_rollback_to_stable.c:1886]: So this is necessary to run in the live database case? My understanding is that we can still skip this for live database

Comment (kommiharibabu) [reply src/txn/txn_rollback_to_stable.c:1886]: We need to run in all the scenarios to clean up any stale entries present in the history store. The actual test format problem occurred running rollback to stable in a live system followed by the update operation.

**Review by agorrod — APPROVED** [review]
LGTM - though this is a high risk bit of code. I've added stress testing into the PR, but since testing had finished we need to check the results manually (they won't be reflected in the PR test status).

**Comment by agorrod** [comment]
Ignore my last comment - the stress tests were already there (thanks). I'm good to go with this.

**Review by quchenhao — APPROVED** [review]
lgtm

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

**Comment by kommiharibabu** [comment]
evergreen merge

**Comment by kommiharibabu** [comment]
The test failures are not related to the PR changes.

**Pull request merged**